src/snowflake/connector/pandas_tools.py

```diff
@@ -120,7 +122,7 @@ def write_pandas(
                 compression, compression_map.keys()
             )
         )
-    if quote_identifiers:
+    if kwargs["quote_identifiers"]:
```

Comment:
I see, I agree with the suggested change.
As for why `kwargs` should be used, my thought process was if we have a similar configuration (like `quote_identifiers`) in the future it would much easier to add it to `default_args` rather than update the signature of `make_pd_writer`, `pd_writer`, and `write_pandas`. `kwargs` handles the argument passing more cleanly in this case (in my opinion).

Reply:
One more small thing, in your suggestion the default value is `False`, should it be `True` (case sensitive by default)

Reply:
> One more small thing, in your suggestion the default value is `False`, should it be `True` (case sensitive by default)

I haven't checked what the default value was, but keep that

Reply:
> I see, I agree with the suggested change.
> As for why `kwargs` should be used, my thought process was if we have a similar configuration (like `quote_identifiers`) in the future it would much easier to add it to `default_args` rather than update the signature of `make_pd_writer`, `pd_writer`, and `write_pandas`. `kwargs` handles the argument passing more cleanly in this case (in my opinion).

Two things:
* Does it really save us any code if we need to keep the dictionary (or `kwargs.get` functions' default values) up to date with the other function instead of the function's arguments?
* So by using `kwargs` you are removing the well-defined nature of the function's argument and default values. Which can be done with good doc-strings (which you just removed in your last push), but we cannot verify this with our automated tools.

I think `kwargs` can be very useful ([here's one place where we need it](https://github.com/snowflakedb/snowflake-connector-python/blob/9098a608e1e20b7f3b0aa43c27e6371101a77143/src/snowflake/connector/ssl_wrap_socket.py#L50)), but it makes documentation a nightmare for those functions, so I'm very against using it with customer facing APIs.

Reply:
I see, that's fair. I'll revert back to using `quote_identifiers` as an argument